src/qiskit_qir/translate.py

```diff
@@ -2,64 +2,126 @@
+        See below
+    :returns: The LLVM bitcode representation of the input.
+
+    :Keyword Arguments:
+        * *record_output* (``bool``) --
+          Whether to record output calls for registers, default `True`
+        * *emit_barrier_calls* (``bool``) --
+          Whether to emit barrier calls in the QIR, default `False`
+    """
+
+    _, module = _build_module(circuits, profile, **kwargs)
+    return module.bitcode
+
+
+def to_qir_bitcode_with_entry_points(
+    circuits: Union[QuantumCircuit, List[QuantumCircuit]],
+    profile: str = "AdaptiveExecution",
+    **kwargs
+) -> Tuple[bytes, List[str]]:
+    r"""Converts the Qiskit QuantumCircuit to QIR bitcode as bytes
+
+    :param circuits:
         Qiskit circuit to be converted to QIR
-    :type circuit: ``QuantumCircuit``
+    :type circuit: ``Union[QuantumCircuit, List[QuantumCircuit]]``
     :param profile:
         The target profile for capability verification
     :type profile: ``str``
     :param \**kwargs:
         See below
+    :returns:
+        Tuple containing the the LLVM bitcode representation of the input and
+        the list of used entry point names generated from the input.
 
     :Keyword Arguments:
         * *record_output* (``bool``) --
           Whether to record output calls for registers, default `True`
-        * *use_static_qubit_alloc* (``bool``) --
-          Whether to use static qubit allocation, default `True`
-        * *use_static_result_alloc* (``bool``) --
-          Whether to use static result allocation, default `True`
+        * *emit_barrier_calls* (``bool``) --
+          Whether to emit barrier calls in the QIR, default `False`
     """
 
-    module = QiskitModule.from_quantum_circuit(circuit=circuit)
-    visitor = BasicQisVisitor(profile, kwargs=kwargs)
-    module.accept(visitor)
-    return visitor.bitcode()
+    entry_points, module = _build_module(circuits, profile, **kwargs)
+    return (module.bitcode, entry_points)
+
+
+def _build_module(
+    circuits: Union[QuantumCircuit, List[QuantumCircuit]],
+    profile: str = "AdaptiveExecution",
+    **kwargs
+) -> Tuple[List[str], Module]:
+    name = "batch"
+    if isinstance(circuits, QuantumCircuit):
+        name = circuits.name
+        circuits = [circuits]
+    elif isinstance(circuits, list):
+        for value in circuits:
+            if not isinstance(value, QuantumCircuit):
+                raise ValueError(
+                    "Input must be Union[QuantumCircuit, List[QuantumCircuit]]"
+                )
+    else:
+        raise ValueError("Input must be Union[QuantumCircuit, List[QuantumCircuit]]")
+
+    if len(circuits) == 0:
+        raise ValueError("No QuantumCircuits provided")
+
+    llvm_module = Module(Context(), name)
+    entry_points = []
+    for circuit in circuits:
+        module = QiskitModule.from_quantum_circuit(circuit, llvm_module)
+        module.accept(BasicQisVisitor(profile, **kwargs))
+        entry_points.append(module.entry_point)
```

Comment:
It seems unnecessary for both the visitor and the module to remember the entry point name. You could do:

```suggestion
        visitor = BasicQisVisitor(profile, **kwargs)
        module.accept(visitor)
        entry_points.append(visitor.entry_point)
```